src/codecs/csl/index.ts

```diff
@@ -186,9 +204,11 @@ export const encodeCreativeWork = (
   logWarnLossIfAny('csl', 'encode', work, lost)
 
   const date = datePublished ?? dateModified ?? dateCreated ?? undefined
+
   return {
     type,
-    id: id !== undefined ? id : crypto.randomBytes(16).toString('hex'),
+    id,
+    'citation-label': id,
```

Comment:
The `citation-label` was added in order to preserve the `reference` ids used in the Markdown files.